### PR TITLE
Add missing tests for Distribution model

### PR DESCRIPTION
**Pull request by saraycp**
Co-authored-by: Moisés Déniz Alemán <mdeniz@suse.com>

### CODE REVIEW

File: src/api/spec/models/obs_factory/distribution_spec.rb [hunk 1]
```diff
@@ -243,5 +243,37 @@ def strategy_for(name)
...
+
+  describe '#openqa_jobs_for' do
+    before do
+      allow(ObsFactory::OpenqaJob).to receive(:find_all_by)
+    end
+
+    shared_examples 'calls find_all_by' do |version, expected_build|
+      it "performs find_all_by with #{version} version" do
+        allow(distribution).to receive("#{version}_version").and_return(expected_build)
+        distribution.openqa_jobs_for(version)
```
Comment (bgeuken) [inline src/api/spec/models/obs_factory/distribution_spec.rb]: I would move those two lines to a before block. Or at least the stub in line 260

File: src/api/spec/models/obs_factory/distribution_spec.rb [hunk 2]
```diff
@@ -243,5 +243,37 @@ def strategy_for(name)
...
+    include_examples 'calls find_all_by', 'published', '20180702'
+    include_examples 'calls find_all_by', 'source', '20180703'
+  end
+
+  describe '#totest_version' do
+    before do
+      allow(distribution.strategy).to receive(:totest_version).and_return('20180702')
+    end
+
+    it { expect(distribution.totest_version).to eq('20180702') }
```
Comment (bgeuken) [inline src/api/spec/models/obs_factory/distribution_spec.rb]: Do we need to test this? It's just delegating the method call to the strategy :thinking:

Comment (saraycp) [reply src/api/spec/models/obs_factory/distribution_spec.rb]: You are right :thinking:  There was another similar test for '#published_version'. Should I remove it as well?

File: src/api/spec/models/obs_factory/distribution_spec.rb (hunk 1 above)
Comment (saraycp) [reply src/api/spec/models/obs_factory/distribution_spec.rb]: Even if each 'allow' is used only in one 'it' block?

Comment (saraycp) [reply src/api/spec/models/obs_factory/distribution_spec.rb]: Well, I tried to put that 'allow' inside the 'before' block and it calls 'allow' 9 times. 3 times per 'it'.
 It is even worst, isn't it?

Comment (hennevogel) [reply src/api/spec/models/obs_factory/distribution_spec.rb]: You know it isn't :smile:

Comment (bgeuken) [reply src/api/spec/models/obs_factory/distribution_spec.rb]: I meant within the shared_example:-)

Like:
```ruby
shared_examples 'calls find_all_by' do |version, expected_build|
  before do
    ...
  end

  it { ... }
end


```

Anyway, this is just a [nitpick](https://github.com/openSUSE/open-build-service/blob/master/CONTRIBUTING.md#how-to-review-code-submissions). You don't have to do it that way. It's just that I find splitting test preparation from the actual test makes them more readable.

Comment (saraycp) [reply src/api/spec/models/obs_factory/distribution_spec.rb]: Yes, I tried with 'before' block inside 'shared_examples'... Well, we think it's better to keep it that way.

**Pull request merged**